eng/packaging.targets

```diff
@@ -22,19 +22,29 @@
                                        '$(IsRIDSpecificProject)' == 'true') and
                                        '$(PreReleaseVersionLabel)' != 'servicing' and
                                        '$(GitHubRepositoryName)' != 'runtimelab'">true</GeneratePackageOnBuild>
+    <GeneratePackageOnBuild Condition="'$(BuildAllConfigurations)' != 'true' and
+                                    '$(IsRIDSpecificProject)' != 'true' and
+                                    '$(PreReleaseVersionLabel)' == 'servicing' and
+                                    '$(GitHubRepositoryName)' != 'runtimelab'">false</GeneratePackageOnBuild>
     <!-- Search for the documentation file in the intellisense package and otherwise pick up the generated one. -->
     <LibIntellisenseDocumentationFilePath>$(XmlDocFileRoot)1033\$(AssemblyName).xml</LibIntellisenseDocumentationFilePath>
     <UseIntellisenseDocumentationFile Condition="'$(UseIntellisenseDocumentationFile)' == '' and Exists('$(LibIntellisenseDocumentationFilePath)')">true</UseIntellisenseDocumentationFile>
   </PropertyGroup>
 
-  <PropertyGroup Condition="'$(ServicingVersion)' != ''">
+  <PropertyGroup Condition="'$(PreReleaseVersionLabel)' == 'servicing'">
+    <!-- If no servicing version is set we need to default to 0 in order for dependency versions to
+    be calculated propertly, if we don't set it to 0, we would get the dependency version using the
+    product Patch Version -->
+    <ServicingVersion Condition="'$(ServicingVersion)' == ''">0</ServicingVersion>
+
     <!-- Always update the package version in servicing. -->
-    <VersionPrefix>$(MajorVersion).$(MinorVersion).$(ServicingVersion)</VersionPrefix>
+    <Version>$(MajorVersion).$(MinorVersion).$(ServicingVersion)</Version>
+    <Version Condition="'$(VersionSuffix)' != ''">$(Version)-$(VersionSuffix)</Version>
```

Comment:
Weren't this things set in arcade before?

Reply:
oh, but we are changing the default since we are using the servicing version, makes sense. I'd probably say ideally we should have that in arcade too as that also makes sense for other repos, but I'm fine with keeping as is on this PR.

Reply:
Yeah, that'd make sense, but I believe other repos don't follow what we do, which is just ship the packages we service, from my understanding other repos ship all the OOB packages, so then this would no longer make sense on arcade. 

cc: @mmitche for thoughts.

Reply:
These are set by arcade in typical scenarios. This may have been related to pkgproj files, which are runtime/mono specific? IIRC the runtime repo had some non-standard behavior that required this be added during the servicing exercise. @Anipik for context.

Reply:
Yes the reason why we needed to add them is because of the way we calculate the package dependencies and because we don't build all packages as part of the build so the version is different for every package we build. In other repos the versions is always Major.Minor.Patch, no matter if the package has fixes or not, so for us we introduced a `ServicingVersion` which is set individually on every serviced package and then the version is `Major.Minor.ServicingVersion`.